packages/twenty-front/src/modules/apollo/services/apollo.factory.ts

```diff
@@ -143,11 +146,18 @@ export class ApolloFactory<TCacheShape> implements ApolloManager<TCacheShape> {
         forward: (operation: Operation) => Observable<FetchResult>,
       ) => {
         return fromPromise(
-          renewToken(uri, getTokenPair())
+          renewToken(uri, getEffectiveTokenPair())
             .then((tokens) => {
               if (isDefined(tokens)) {
-                onTokenPairChange?.(tokens);
-                cookieStorage.setItem('tokenPair', JSON.stringify(tokens));
+                if (isImpersonating()) {
```

Comment:
I'm not a fan of the naming: utils are meant to be called synchronously:
- if they are called in the reactive part, they will trigger re-renders

So they should have a name that suggest that they are doing sync action: doSomething()

"isImperosonating" is suggesting this is reactive.

I would call it: getIsImpersonating()